src/models/booking.cr

```diff
@@ -79,6 +140,10 @@ class Booking
     add_error("booking_end", "must be after booking_start") if booking_end <= booking_start
   end
 
+  private def validate_history_size
+    add_error("history", "must contain at most 3 events") if history.size > 3
```

Comment:
@caspiano
So, change this from validation to a warning log?

Reply:
Assuming `history` cannot be assigned by the user, this is a server failure and it should be an error log.
Try to make some unit tests that validate it's not possible to get into this state.

Reply:
I have changed it to an error log and made sure that #create and #update can't set the history.
The size from various normal state transitions is tested as part of "#show should include history of state changes".

At this point, I can't think of any way to make it be bigger without bypassing the controller.